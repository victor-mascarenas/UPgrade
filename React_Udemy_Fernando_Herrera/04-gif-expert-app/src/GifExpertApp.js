import React, {useState} from "react";

const GifExpertApp = () => {
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball', 'Naruto']
    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball', 'Naruto']);
    
    const addButtonOnClick = () => {
        const newItem = 'Family Guy';
        setCategories([...categories, newItem]);
    };

    return(
        <>
            <h2>GifExpertApp</h2>
            <hr/>
            <button onClick={addButtonOnClick}>Agregar</button>
            <ul>
                {
                    categories.map((category, key) => {
                        return <li key={`${category}_${key}`}>{category}</li>;
                    })
                }
            </ul>
        </>
    );
}

export default GifExpertApp;