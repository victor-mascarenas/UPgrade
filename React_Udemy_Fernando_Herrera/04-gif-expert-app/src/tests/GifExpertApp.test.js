import React from "react";
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente GifExpertApp', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<GifExpertApp/>);
    });
    
    test('Debe empatar con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One punch', 'Dragon ball'];
        wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});
