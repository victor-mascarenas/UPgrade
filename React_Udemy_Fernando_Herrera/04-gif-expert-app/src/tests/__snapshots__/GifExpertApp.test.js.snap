// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch_0"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball_1"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente GifExpertApp Debe empatar con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
